evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -82,17 +85,22 @@ def _get_cat_cols(X):
 
     def fit(self, X, y=None):
         top_n = self.parameters['top_n']
-        if not isinstance(X, pd.DataFrame):
-            X = pd.DataFrame(X)
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
         X_t = X
 
         if self.features_to_encode is None:
             self.features_to_encode = self._get_cat_cols(X_t)
         invalid_features = [col for col in self.features_to_encode if col not in list(X.columns)]
         if len(invalid_features) > 0:
             raise ValueError("Could not find and encode {} in input data.".format(', '.join(invalid_features)))
-
         if self.parameters['handle_missing'] == "as_category":
+            for col in self.features_to_encode:
```

Comment:
Syntax using `where`: https://stackoverflow.com/questions/36825925/expressions-with-true-and-is-true-give-different-results

Reply:
Yup, because the new nullable boolean--which is hard to test for equality--we need to grab these values and convert them to the "nan" category manually.

Reply:
Ah neat.

Does this work?
```python
X_t[col] = X_t[col].where(~pd.isna(X_t[col]), other='nan')
```
[The tilde is the bitwise negation operator in pandas](https://stackoverflow.com/questions/46054318/tilde-sign-in-python-dataframe)

Reply:
Oo yes, I always forget this exists. Thanks, updating since this is cleaner!